tutorials/arvo/claz.md

```diff
@@ -0,0 +1,72 @@
+
+First you will want to boot a fakezod. You can do this with your live running ship but at this time this will need you to modify the code so it's best to do this on a ship you can make mistakes on. If you don't know how to boot a fakezod see [this guide](https://urbit.org/using/develop/#creating-a-development-ship) for instructions.
+
+Be sure to have mounted your home desk so you can easily edit files on it.
+The first edit that will probably need to be made is in [`app/claz.hoon`](https://github.com/urbit/urbit/blob/master/pkg/arvo/app/claz.hoon#L14). The listed url will need to be changed to a valid Ethereum endpoint. infura.io provides an Ethereum API that you can use.
+
+The second change that may potentially be needed is to modify the [gas limit](https://github.com/urbit/urbit/blob/master/pkg/arvo/app/claz.hoon#L179). This number here is the maximum number of gas units that will be used. In future this will be configurable but for now you may need to change this manually. For a guide to what to set it to you can use the [constants in bridge](https://github.com/urbit/bridge/blob/master/src/lib/constants.js#L23) as a guide based on what action you are going to be performing. The maximum cost of your transaction will be the maximum gas units multiplied by the gwei you're going to set later. You'll need at least this much in your account.
+
+Once you have made these changes be sure to run `|commit %home` to get them into your ship.
+
+Next run `|start %claz` after which you'll be able to generate transactions.
+
+```
+:claz [%generate %/spawn/eth-txs %mainnet author %single %spawn ~ship to]
+```
+
+Here we have an example of generating a transaction to spawn a ship. Something to note is that for this to be successful you'll need to actually have the right to spawn the ship in question. Let's break down the command.
+
+
+`%generate` is how all commands to `claz` are going to start.
+
+`%/spawn/eth-txs` is simply the path for the output file
+
+`%mainnet` specifies which network this is going to sent out on. Other options are possible including the test-net `%ropsten`.
+
+`author` here is an Ethereum address that is issuing the transaction. It will need to be written in `@ux` notation, e.g. `0x3b17.d097.d9dd.711e.4ef8.517a.bbf1.8b2b.a643.81fe`.
+
+`%single` specifies what kind of batch that will be created. If you are going to do multiple transactions you will need to either set this to `%more` and create a list of batches or upload them before generating the next one. The reason this ship needs to have internet access is because it needs to look up the nonce for the specified address.
+
+`%spawn` is the command we want to issue. After this what the arguments are will vary based on the command.
+
+`~ship` will be the ship to be spawned.
+
+`to` is the Ethereum address to spawn the ship to.
+
+You will want to read `sur/claz.hoon` which you can find [here](https://github.com/urbit/urbit/blob/master/pkg/arvo/sur/claz.hoon).
```

Comment:
```suggestion
You will want to read [`/sur/claz.hoon`](https://github.com/urbit/urbit/blob/master/pkg/arvo/sur/claz.hoon).
```

Looking at that file with uninitiated eyes, `%single` and `%more` are about the only things I can easily understand. And in truth, even the `+call` holds some secrets that might bite you (`%transfer-ship` never resets keys/proxies, for example). More elaboration may or may not be warranted here, but I guess most of these are "you probably won't ever need this".

Reply:
Eh. I kind of think the fact that it's not ultra easy to read is a bonus for this guide because people shouldn't be doing this unless they are _very_ confident.